### PR TITLE
Add ZMS to mesh samples. Remove nrf54l15pdk overlays

### PR DESCRIPTION
**Pull request by akredalen**
Discussion:
* Other teams have added a Kconfig to each of their samples to generalize ZMS for all devices that contain RRAM (such as the nRF54L Series), and then removed `CONFIG_NVS=y` from the prj.conf. See the PR [here](https://github.com/nrfconnect/sdk-nrf/pull/17502/files#diff-824da088984ec8a143b58288f5a29969bc4fdf5f8e102a1bb9c3ffed7068f2b3). Is this something we want to do as well? Currently, this PR only adds ZMS if a sample is compiled with the 54L**15** (as according to the JIRA task) to already existing files.

All samples have been tested (provisioned).

Removed nrf54L15pdk overlays as this seems to have been forgotten in previous [PR](https://github.com/nrfconnect/sdk-nrf/pull/17913).

TODO:
- [x] Review recommendations for samples using ZMS through Settings

### CODE REVIEW

**Comment by ludvigsj** [comment]
IMO, since these samples only support 1 RRAM-based device currently (nRF54L15), and that device already has a board conf in each sample, it'll be premature overcomplication to start redefining Kconfig options to select ZMS based on presence of RRAM right now. I like your current solution. We can keep the Kconfig override in mind when/if we start adding support for more RRAM-based devices in the future, adding to the board conf is the best solution at the current moment.

File: samples/bluetooth/mesh/ble_peripheral_lbs_coex/boards/nrf54l15dk_nrf54l15_cpuapp.conf [hunk 1]
```diff
@@ -7,3 +7,7 @@
 # Application overlay - nrf54l15
 
 CONFIG_SOC_FLASH_NRF_PARTIAL_ERASE=n
+CONFIG_NVS=n
+CONFIG_NVS_LOOKUP_CACHE=n
+CONFIG_SETTINGS_NVS_NAME_CACHE=n
+CONFIG_ZMS=y
```
Comment (Balaklaka) [inline samples/bluetooth/mesh/ble_peripheral_lbs_coex/boards/nrf54l15dk_nrf54l15_cpuapp.conf:13]: Maybe we should add these too for all the samples:
CONFIG_SETTINGS_ZMS_SECTOR_COUNT=8
CONFIG_ZMS_LOOKUP_CACHE=y
CONFIG_ZMS_LOOKUP_CACHE_SIZE=512
CONFIG_SETTINGS_ZMS_NAME_CACHE=y
CONFIG_SETTINGS_ZMS_NAME_CACHE_SIZE=512
CONFIG_ZMS_LOOKUP_CACHE_FOR_SETTINGS=y

File: doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst [hunk 2]
```diff
@@ -471,6 +471,8 @@ Bluetooth Mesh samples
   * :ref:`ble_mesh_dfu_target`
   * :ref:`ble_mesh_dfu_distributor`
 
+* Replaced usage of NVS with the ZMS file subsystem when compiling samples for nRF54L15.
```
Comment (Balaklaka) [inline doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst:474]:
```suggestion
* Added support for ZMS file subsystem when compiling for :ref:`zephyr:nrf54l15dk_nrf54l15` in all samples supporting that board.
```

**Review by Balaklaka — CHANGES_REQUESTED** [review]

File: samples/bluetooth/mesh/ble_peripheral_lbs_coex/boards/nrf54l15dk_nrf54l15_cpuapp.conf (hunk 1 above)
Comment (akredalen) [reply samples/bluetooth/mesh/ble_peripheral_lbs_coex/boards/nrf54l15dk_nrf54l15_cpuapp.conf:13]: Where did you get these values? Are these based of the testing that Omkar did perhaps?

**Review by omkar3141 — APPROVED** [review]

**Review by Balaklaka — APPROVED** [review]

File: samples/bluetooth/mesh/light/boards/nrf54l15dk_nrf54l15_cpuapp.conf [hunk 3]
```diff
@@ -0,0 +1,20 @@
+#
+# Copyright (c) 2024 Nordic Semiconductor ASA
+#
+# SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+#
+################################################################################
+# Application overlay - nrf54l15
+
+CONFIG_SOC_FLASH_NRF_PARTIAL_ERASE=n
+CONFIG_SPI_NOR=n
```
Comment (HaavardRei) [inline samples/bluetooth/mesh/light/boards/nrf54l15dk_nrf54l15_cpuapp.conf:10]: I see this is not your change, but why is this disabled only for this specific sample?

File: doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst [hunk 4]
```diff
@@ -471,6 +471,9 @@ Bluetooth Mesh samples
   * :ref:`ble_mesh_dfu_target`
   * :ref:`ble_mesh_dfu_distributor`
 
+* Added support for ZMS file subsystem when compiling for :ref:`zephyr:nrf54l15dk_nrf54l15` in all
+  samples supporting that board.
+
```
[the remark this answers is not in the document]
Comment (akredalen) [reply doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst:476]: The removed support for the nRF54L15 PDK was actually done in a different PR, [here](https://github.com/nrfconnect/sdk-nrf/pull/17913/files#diff-9460bd2e76b8fe433e47a738942f77d38f18d5967374101c756a015372b8cc22). Only some parts was left out

[the remark this answers is not in the document]
Comment (akredalen) [reply doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst]: Fixed.

**Review by mia-ko — APPROVED** [review]

**Review by divipillai — APPROVED** [review]

**Pull request merged**